pkg/reconciler/workload/namespace/namespace_reconcile.go

```diff
@@ -172,7 +172,7 @@ func (c *Controller) reconcileGVR(ctx context.Context, gvr schema.GroupVersionRe
 // will succeed without error if a cluster is assigned or if there are no viable clusters
 // to assign to. The condition of not being scheduled to a cluster will be reflected in
 // the namespace's status rather than by returning an error.
-func (c *Controller) ensureScheduled(ctx context.Context, ns *corev1.Namespace) (*corev1.Namespace, error) {
+func (c *Controller) ensureScheduled(ctx context.Context, ns *corev1.Namespace) (*corev1.Namespace, bool, error) {
```

Comment:
I would suggest using named return variables so it's clear what this bool is. Also would update the godoc to add it.

Alternatively (follow-up), refactor this so it's not assigning + patching + returning true/false all in one method?